### PR TITLE
Fixup Kubernetes examples and docs

### PR DESCRIPTION
**Pull request by udnay**
The current way that we create certificates for CRDB to perform mTLS between nodes on Kubernetes is being deprecated. We should update the documentation to reflect this. Additionally our examples for connecting and initializing  a secure cluster do not work with the operator. This PR is updating the documentation as well as removing old and unsupported versions of Kubernetes.

### CODE REVIEW

File: cloud/kubernetes/README.md [hunk 1]
```diff
@@ -25,11 +25,16 @@ kubectl run cockroachdb --image=cockroachdb/cockroach --restart=Never -- start-s
...
-Older Kubernetes versions that don't have their own directory are no longer
-supported.
+directory without modification is `1.18`.
+
+
+### Secure Mode
+
+The use of Kubernetes certificate authority to sign CockroachDBs certificates has been deprecated by Kubernetes. Due to 
+the deprecation to enable secure mode we currently only support the `bring-your-own-certs` configuration. 
+The deprecated method for using the Kubernetes signing authority is in the `v1.17` directory.
```
Comment (pseudomuto) [inline cloud/kubernetes/README.md:35]: Didn't you delete the v1.17 dir? Did you mean the kubernetes-signer dir?

File: cloud/kubernetes/bring-your-own-certs/cluster-init.yaml [hunk 2]
```diff
@@ -16,4 +16,13 @@ spec:
...
           - "--insecure"
           - "--host=cockroachdb-0.cockroachdb"
+        volumeMounts:
+          - name: client-certs
+            mountPath: /cockroach-certs
       restartPolicy: OnFailure
+      volumes:
+        - name: client-certs
+          secret:
+            secretName: cockroachdb.client.root
```
Comment (pseudomuto) [inline cloud/kubernetes/bring-your-own-certs/cluster-init.yaml:26]: I think we should let people know they need to create this secret. Or am I missing something that auto-creates it?

**Review by taroface — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72275)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pseudomuto, @taroface, and @udnay)

___
*[cloud/kubernetes/README.md, line 33 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72275#-MnXJL8e9saa5r1lFW8u:-MnXJL8f3Lxrw3vDcfo9:b1tmoiy) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0e958641947547e50f9e3be9b728527aa162b1d/cloud/kubernetes/README.md#L33)):*
> ```
> CockroachDBs certificates
> ```

Nit: "certificates for CockroachDB"
___
*[cloud/kubernetes/README.md, line 33 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72275#-MnXK1tc-kQpBRhnpe9S:-MnXK1tc-kQpBRhnpe9T:b-t6w7mi) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0e958641947547e50f9e3be9b728527aa162b1d/cloud/kubernetes/README.md#L33)):*
> ```
> has been deprecated by Kubernetes.
> ```

Should we add "as of 1.18" (if that's accurate)?
___
*[cloud/kubernetes/README.md, line 34 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72275#-MnXJdmA--wy8Cs9EtQh:-MnXJdmA--wy8Cs9EtQi:b77l1eh) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0e958641947547e50f9e3be9b728527aa162b1d/cloud/kubernetes/README.md#L34)):*
> ```
> Due to the deprecation
> ```

Nit: I'd delete the above for clarity (starting the sentence with "To enable secure mode,")
___
*[cloud/kubernetes/README.md, line 37 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72275#-MnXJrnk6dQZYdMgGICs:-MnXJrnk6dQZYdMgGICt:bort51w) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0e958641947547e50f9e3be9b728527aa162b1d/cloud/kubernetes/README.md#L37)):*
> ```
> Signing
> ```

Should this be "Certificate"?
___
*[cloud/kubernetes/kuberenetes-signer/README.md, line 1 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/72275#-MnXX7UX52jbQe_GzN8j:-MnXX7UX52jbQe_GzN8k:bcmcq6p) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd87b251611bd825f34924926596d41cdf5ef4bb/cloud/kubernetes/kuberenetes-signer/README.md#L1)):*
> ```Markdown
> ### Instructions for using the Kubernetes Self-Signed Certificate to deploy CockroachDB
> ```

Maybe add a note that this represents the deprecated flow?

`Note: This method for using the Kubernetes signing authority has been deprecated. For the current method, see cloud/kubernetes/README.md`.
___
*[cloud/kubernetes/bring-your-own-certs/cluster-init.yaml, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72275#-MnOiTKLbS0tyALen0fb-r1-26:-MnXVENb86uwj2-BvUbY:b-3q2reb) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0e958641947547e50f9e3be9b728527aa162b1d/cloud/kubernetes/bring-your-own-certs/cluster-init.yaml#L26)):*
<details><summary><i>Previously, pseudomuto (David Muto) wrote…</i></summary><blockquote>

I think we should let people know they need to create this secret. Or am I missing something that auto-creates it?
</blockquote></details>

FYI we only use this yaml on our insecure deployment docs. I'm not quite sure why or whether this makes the most sense.

Insecure steps: https://www.cockroachlabs.com/docs/v21.2/deploy-cockroachdb-with-kubernetes-insecure#step-2-start-cockroachdb
 
The secure steps have us execing into a pod and running `cockroach init`: https://www.cockroachlabs.com/docs/stable/deploy-cockroachdb-with-kubernetes.html?filters=manual#step-2-start-cockroachdb

File: cloud/kubernetes/README.md (hunk 1 above)
Comment (taroface) [reply cloud/kubernetes/README.md:35]: +1

**Pull request closed without merging**